cas-mfa-java/src/main/java/net/unicon/cas/mfa/web/support/AbstractMultiFactorAuthenticationArgumentExtractor.java

```diff
@@ -75,9 +77,12 @@ public final WebApplicationService extractService(final HttpServletRequest reque
         authenticationMethod = this.authenticationMethodTranslator.translate(targetService, authenticationMethod);
         this.authenticationMethodVerifier.verifyAuthenticationMethod(authenticationMethod, targetService, request);
 
+        // Grab the HTTP method for the response off of the request.
+        final String method = request.getParameter(CONST_PARAM_METHOD);
+
         final MultiFactorAuthenticationSupportingWebApplicationService mfaService =
                 this.mfaWebApplicationServiceFactory.create(targetService.getId(), targetService.getId(), targetService.getArtifactId(),
-                        authenticationMethod, getAuthenticationMethodSource());
+                        "POST".equals(method) ? ResponseType.POST : ResponseType.REDIRECT, authenticationMethod, getAuthenticationMethodSource());

```

Comment:
It doesn't appear that a `getResponseType()` method exists. There's a `getResponse()`.